src/resources/platform_handlers/pytorch_libtorch/model.py

```diff
@@ -0,0 +1,241 @@
+# CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+# PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+# OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import importlib
+import json
+import os
+
+try:
+    import torch
+except ModuleNotFoundError as error:
+    raise RuntimeError(
+        "Missing/Incomplete PyTorch package installation... (Did you install PyTorch?)"
+    ) from error
+
+# triton_python_backend_utils is available in every Triton Python model. You
+# need to use this module to create inference requests and responses. It also
+# contains some utility functions for extracting information from model_config
+# and converting Triton input/output types to numpy types.
+import triton_python_backend_utils as pb_utils
+
+
+def _get_model_path(config):
+    filenames = ["model.py", "model.pt", "model.pth"]
+    if config["default_model_filename"]:
+        filenames.insert(0, config["default_model_filename"])
+    for filename in filenames:
+        model_path = os.path.join(pb_utils.get_model_dir(), filename)
+        if os.path.exists(model_path):
+            return model_path
+    raise pb_utils.TritonModelException(
+        "No model found in " + pb_utils.get_model_dir() + "/" + str(filenames)
+    )
+
+
+def _get_model_data_path(model_path):
+    data_path_extensions = [".pt", ".pth"]
+    for extension in data_path_extensions:
+        data_path = model_path + extension
+        if os.path.exists(data_path):
+            return data_path
+    # data file not provided
+    return ""
+
+
+def _is_py_class_model(model_path):
+    return model_path[-3:] == ".py"
+
+
+def _import_module_from_path(module_name, file_path):
+    spec = importlib.util.spec_from_file_location(module_name, file_path)
+    module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(module)
+    return module
+
+
+def _get_model_class_from_module(module):
+    names = dir(module)
+    for name in names:
+        attr = getattr(module, name)
+        try:
+            if issubclass(attr, torch.nn.Module):
+                return attr
+        except TypeError:
+            # attr may not be a class
+            pass
+    raise pb_utils.TritonModelException("Cannot find a subclass of torch.nn.Module")
+
+
+def _parse_io_config(io_config):
+    io = []
+    for conf in io_config:
+        io.append({"name": conf["name"]})
+    return io
+
+
+def _get_device_name(model_instance_kind, model_instance_device_id):
+    if model_instance_kind == "GPU":
+        return "cuda:" + model_instance_device_id
+    return "cpu"
+
+
+def _enable_torch_compile(config):
```

Comment:
You should probably expose config parameters to set parallelism. 
https://pytorch.org/docs/stable/torch.html#parallelism
[set_num_threads](https://pytorch.org/docs/stable/generated/torch.set_num_threads.html#torch.set_num_threads)
[set_num_threads](https://pytorch.org/docs/stable/generated/torch.set_num_threads.html#torch.set_num_threads)
From our past experience the default value should be 1.

Reply:
The ability to set those values is added.

From my testing, both values are defaulting to 32.  Should we leave the default value as it is, if the user is not providing the setting?

The values used for each model instance can be viewed by setting `log-verbose=1+` with/after [Enable setting torch parallelism parameters](https://github.com/triton-inference-server/python_backend/pull/282/commits/899dd7345b82d6f68b047648473d3b61b7c49841) commit.

Reply:
I would advice setting the default value to be 1. Scaling the number of instances might quickly cause over-subscription of CPU otherwise.

Reply:
Yes. [Use default value 1 for torch parallelism](https://github.com/triton-inference-server/python_backend/pull/282/commits/26bd52a50cd78e554ad6cec3f08b8194cf438741)

Reply:
and in general letting the user provide environment variables might be nice for pytorch, as sometimes need to pass extra parallelization configs to openmp, and not always pytorch allows to control them via calling python methods...

Reply:
Interesting... will keep an eye out for possible issues, as we are only trying to expose some PyTorch configs at this time.